Annotate VideoPresentationManagerProxy endpoints with feature flag

darryl-apple opened the pull request:
#### 19b4684c588f2eb373cf9fd6c8ec7d87252afc6c
<pre>
Annotate VideoPresentationManagerProxy endpoints with feature flag
<a href="https://bugs.webkit.org/show_bug.cgi?id=283916">https://bugs.webkit.org/show_bug.cgi?id=283916</a>
<a href="https://rdar.apple.com/140798582">rdar://140798582</a>

Reviewed by Andy Estes.

Annotate VideoPresentationManagerProxy endpoints with feature flag

* Source/WebKit/UIProcess/Cocoa/VideoPresentationManagerProxy.messages.in:

Canonical link: <a href="https://commits.webkit.org/288653@main">https://commits.webkit.org/288653@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/aed9aeb5f38c8aadefba7d5154eb116cade6eac5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/83871 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/3489 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/38172 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/88945 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/34879 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/3580 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/11456 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/65235 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/23070 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/86917 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/2649 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/76205 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/45527 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/2578 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/30426 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/33928 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/76834 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/73575 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/31180 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/90321 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/82888 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/11136 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/8071 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/73680 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/11360 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/72034 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/72899 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18054 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/17179 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/15863 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/2466 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/11088 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/16560 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/105306 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/10936 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/25449 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/14412 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/12708 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aestes reviewed (changes requested):
I don't think this is correct on visionOS, where `VideoPresentationModeAPIEnabled` is off but `VideoPresentationManagerProxy` is still used.

aestes reviewed (changes requested):
Instead of duplicating the message signatures, can we instead add a new preference (in UnifiedWebPreferences.yaml) that is enabled on visionOS, then do `EnabledBy = VideoPresentationModeAPIEnabled || NewFeatureEnabled`?

aestes commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -7229,6 +7229,21 @@ SpatialVideoEnabled:
     WebCore:
       default: false
+      
+SpatialVideoFullscreenEnabled:
```
Nit: spatial video has a specific meaning on visionOS (stereoscopic video), but VideoPresentationManagerProxy is used for monoscopic video as well on visionOS. Perhaps a better name would be `VideoPresentationManagerEnabled`.

aestes commented on Source/WebKit/UIProcess/Cocoa/VideoPresentationManagerProxy.messages.in:
```diff
@@ -21,11 +21,9 @@
-[
-    ExceptionForEnabledBy,
-    DispatchedFrom=WebContent,
-    DispatchedTo=UI
```
Just confirming that you intended to remove `DispatchedFrom` and `DispatchedTo`.

darryl-apple replied:
I did not, thanks for flagging. Fixed

aestes reviewed (approved):
r=me with a nit and a question.

darryl-apple committed:
Annotate VideoPresentationManagerProxy endpoints with feature flag
https://bugs.webkit.org/show_bug.cgi?id=283916
rdar://140798582

Reviewed by Andy Estes.

Annotate VideoPresentationManagerProxy endpoints with feature flag

* Source/WebKit/UIProcess/Cocoa/VideoPresentationManagerProxy.messages.in:

Canonical link: https://commits.webkit.org/288653@main
